stl/inc/xlocale

```diff
@@ -200,10 +200,10 @@ public:
         static void __CLRCALL_OR_CDECL _Makewloc(
             const _Locinfo&, category, _Locimp*, const locale*); // make wchar_t facets
 
-#ifdef _NATIVE_WCHAR_T_DEFINED
+#if defined(_NATIVE_WCHAR_T_DEFINED) && !_ENFORCE_FACET_SPECIALIZATIONS

```

Comment:
I believe that additional changes are necessary for completeness. Currently, when enforcing facet specializations, we sometimes permit `unsigned short`. Examples:

https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/locale#L26-L30
https://github.com/microsoft/STL/blob/17fde2cbab6e8724d81c9555237c9a623d7fb954/stl/inc/xlocmes#L27-L31

(This is not an exhaustive list, but all occurrences can be found by searching for `!_ENFORCE_FACET_SPECIALIZATIONS`.)

I believe that this PR should now remove `unsigned short` whenever it appears in these assertions:

* When we aren't enforcing facet specializations, everything is allowed.
* When we are enforcing:
  + Users with real `wchar_t` should not be permitted to use `unsigned short` - the only reason it was mentioned here was because the STL itself was unconditionally doing so, which is what this PR is fixing.
  + Users with fake `wchar_t` will be permitted to use fake `wchar_t`, i.e. `unsigned short` - there is no reason to mention the same type in two different ways.